src/Parsers/parseQuery.cpp

```diff
@@ -241,70 +250,90 @@ ASTPtr tryParseQuery(
         //"
         // Advance the position, so that we can use this parser for stream parsing
         // even in presence of such queries.
-        pos = token_iterator->begin;
+        _out_query_end = token_iterator->begin;
         return nullptr;
     }
 
     Expected expected;
-
     ASTPtr res;
-    bool parse_res = parser.parse(token_iterator, res, expected);
-    Token last_token = token_iterator.max();
+    const bool parse_res = parser.parse(token_iterator, res, expected);
+    const auto last_token = token_iterator.max();
+    _out_query_end = last_token.end;
 
-    /// If parsed query ends at data for insertion. Data for insertion could be in any format and not necessary be lexical correct.
     ASTInsertQuery * insert = nullptr;
     if (parse_res)
         insert = res->as<ASTInsertQuery>();
 
-    if (!(insert && insert->data))
+    // If parsed query ends at data for insertion. Data for insertion could be
+    // in any format and not necessary be lexical correct, so we can't perform
+    // most of the checks.
+    if (insert && insert->data)
     {
-        /// Lexical error
-        if (last_token.isError())
+        if (!parse_res)
```

Comment:
@akuzm, it's always false

Reply:
Oh.... So how does it manifest, we display bad messages for broken inserts? I have to add some test for this.

Reply:
Looks like we can just remove it: https://github.com/ClickHouse/ClickHouse/pull/19919